Sp 78

PTsiali opened the pull request:
Review of SP-78

FelixDittrich committed:
First Commit after rebase

FelixDittrich committed:
First Commit after rebase

PTsiali committed:
[SP-78] Added page Exam-Archiv

PTsiali committed:
[SP-78] Added page Exam-Archiv

[SP-78] Added page Exam-Archiv 2

[SP-78] Added page Exam-Archiv 2

[SP-78] Small changes of variable name, comments added and changed

PTsiali committed:
[SP-78] Added page Exam-Archiv 2

PTsiali committed:
[SP-78] Added page Exam-Archiv 2

MediocreMarcel commented on Backend/src/main/java/de/hft_stuttgart/winf/proj2/sp/backend/db_access/DbExam.java:
```diff
@@ -65,8 +65,31 @@ public boolean createExams(CreateExamDto exam) throws SQLException {
     }
-}
 
 
```
Add Javadoc Comment

MediocreMarcel commented on Backend/src/main/java/de/hft_stuttgart/winf/proj2/sp/backend/dto/ExamDto.java:
```diff
@@ -28,9 +28,12 @@
     @Column("module_id")
     private Integer moduleId;
+    @Column("total_points")
+    private Integer total_points;
```
Name Variable totalPoints.
Leave @Column as it is

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/archived-exams/archived-exams-datasource.ts:
```diff
@@ -0,0 +1,96 @@
+import { Observable, of as observableOf, merge } from 'rxjs';
+import {ExamDTO} from "../models/ExamDTO";
+
+// TODO: Replace this with your own data model type
```
Remove todo if no longer needed

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/archived-exams/archived-exams.component.ts:
```diff
@@ -0,0 +1,58 @@
+
+  }
+
+  loadExams() {
```
Add /** comment

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/archived-exams/archived-exams.component.ts:
```diff
@@ -0,0 +1,58 @@
+
+  }
+
+  loadDataIntoTable(){
```
add /** comment

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/models/ExamDTO.ts:
```diff
@@ -7,14 +7,16 @@ export class ExamDTO {
   creation_date: string ;
   exam_date: number;
   status: any;
+  total_points: number;
```
change to totalPoints

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/models/ExamDTO.ts:
```diff
@@ -7,14 +7,16 @@ export class ExamDTO {
   module: ModuleDTO;
 
-  constructor(exam_id,title,creation_date,exam_date,status,module) {
+  constructor(exam_id,title,creation_date,exam_date,status,total_points,module) {
```
change to totalPoints

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/models/ExamDTO.ts:
```diff
@@ -7,14 +7,16 @@ export class ExamDTO {
     this.creation_date = creation_date;
     this.exam_date = exam_date;
     this.status = status;
+    this.total_points= total_points;
```
change to totalPoints

MediocreMarcel commented on Frontend/SPFrontend/src/app/services/exam-archiv/exam-archive-service.service.ts:
```diff
@@ -0,0 +1,21 @@
+
+  constructor(private http: HttpClient) { }
+
+  getExamsFromArchivForUser(user: User){
```
Add /** comment

MediocreMarcel reviewed (changes requested):
Fix feedback

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/archived-exams/archived-exams.component.ts:
```diff
@@ -0,0 +1,58 @@
+  displayedColumns = ['exam_id', 'title', 'creation_date', 'exam_date', 'total_points',  ];
+
+
+  public EXAMPLE_DATA: ExamDTO[];
```
Variablen name nicht example

MediocreMarcel replied:
Variablen name nicht example

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/archived-exams/archived-exams.component.ts:
```diff
@@ -0,0 +1,58 @@
+  constructor(private service: ExamArchiveServiceService, private userService: UserService ) {
+    this.loadExams();
+
+    console.log(this.EXAMPLE_DATA);
```
Console log raus

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/archived-exams/archived-exams.component.ts:
```diff
@@ -0,0 +1,58 @@
+
+
+       this.EXAMPLE_DATA = u;
+      this.dataSource = new ArchivedExamsDataSource(this.EXAMPLE_DATA);
```
Variablen name nicht example

MediocreMarcel commented on Frontend/SPFrontend/src/app/components/archived-exams/archived-exams-datasource.ts:
```diff
@@ -0,0 +1,96 @@
+}
+
+
+/** Simple sort comparator for example ID/Name columns (for client-side sorting). */
```
kommentar überarbeiten

MediocreMarcel committed:
Merge branch 'main' into SP-78

MediocreMarcel committed:
fix navbar

MediocreMarcel committed:
Merge branch 'main' of https://github.com/MediocreMarcel/SP into main

FelixDittrich committed:
now able to add and remove questions

FelixDittrich committed:
now able to add and remove questions

MediocreMarcel committed:
fix animations

FelixDittrich committed:
resolved changerequest

FelixDittrich committed:
resolved changerequest

FelixDittrich committed:
added comments

FelixDittrich committed:
added comments

MediocreMarcel committed:
[SP-77] Code Adaptions for DB Change.
 ResultSetMapper had to undergo mayor changes to support object in object mapping. New DTO and adaption of old dtos

MediocreMarcel committed:
[SP-77] DB Changes

MediocreMarcel committed:
Merge remote-tracking branch 'origin/QuestionTest' into QuestionTest

# Conflicts:
#	Frontend/SPFrontend/src/app/app.module.ts

MediocreMarcel committed:
Merge pull request #32 from MediocreMarcel/QuestionTest

add and remove questions from questionpool

Commit:
Merge pull request #33 from MediocreMarcel/SP-77

Sp 77

MediocreMarcel committed:
[SP-76] fix order of examquestions, fix course selection, add save notification, fix html view

MediocreMarcel committed:
[SP-76] Add SQL Files of fix

arthi1997 committed:
Merge pull request #34 from MediocreMarcel/SP-76-Fixes

Sp 76 fixes

arthi1997 committed:
[SP-82] - Keep User in Session

MediocreMarcel committed:
fix db connection problem

MediocreMarcel committed:
Merge pull request #35 from MediocreMarcel/SP-82

[SP-82] - Keep User in Session

PTsiali committed:
Merge remote-tracking branch 'origin/SP-78' into SP-78

# Conflicts:
#	Backend/src/main/java/de/hft_stuttgart/winf/proj2/sp/backend/db_access/DbExam.java
#	Frontend/SPFrontend/src/app/app.module.ts
#	Frontend/SPFrontend/src/app/components/archived-exams/archived-exams-datasource.ts
#	Frontend/SPFrontend/src/app/components/archived-exams/archived-exams.component.html
#	Frontend/SPFrontend/src/app/components/archived-exams/archived-exams.component.ts

PTsiali committed:
[SP-78] Duoble method deleted

MediocreMarcel reviewed (approved):
LGTM
